osf/management/commands/backfill_domain_references.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+import re
+import logging
+import operator
+from functools import reduce
+
+from django.apps import apps
+from django.db.models import Q
+from django.core.management.base import BaseCommand
+from framework.celery_tasks import app as celery_app
+from osf.models import NotableDomain, DomainReference
+from django.contrib.contenttypes.models import ContentType
+
+logger = logging.getLogger(__name__)
+
+DOMAIN_MATCH_REGEX = re.compile(r'(?P<protocol>\w+://)?(?P<www>www\.)?(?P<domain>[\w-]+\.\w+)(?P<path>/\w*)?')
+DOMAIN_SEARCH_REGEX = r'(http://[^ \'}\[\]\~\(\)\/]+|https://[^ \'}\[\]\~\(\)\/]+)'
+
+
+@celery_app.task()
+def create_notable_domain_with_reference(domain, resource_id, resource_content_type_pk):
+    domain, created = NotableDomain.objects.get_or_create(
+        domain=domain,
+        defaults={'note': NotableDomain.Note.UNKNOWN}
+    )
+    DomainReference.objects.get_or_create(
+        domain=domain,
+        referrer_object_id=resource_id,
+        referrer_content_type_id=resource_content_type_pk
+    )
+
+    if created:
+        logger.info(f'Creating NotableDomain {domain}')
```

Comment:
I don't understand how you are going to find new domain instances in that case, I think we are hazy about the use case here, we should talk about this during our 1-on-1 today.

Reply:
Firing off a task just to create a DomainReference is not effective use of resources because it means that the work to determine which DomainReferences to create is all still being done in one long loop. It also means you're potentially creating a lot more tasks, which bogs down the queue.

The more efficient thing would be to fire off a task to extract and create DomainReferences from each thing that might have one. Then you're just running through a list of IDs and creating tasks to handle the extraction and creation, meaning the the individual extraction tasks are independent of each other.

As suggested below, if you can clean up the `_get_spam_content` function, then you can literally just reuse the task that your'e using a part of the `check_spam` call.